rts/Sim/MoveTypes/GroundMoveType.cpp

```diff
@@ -518,6 +518,8 @@ CGroundMoveType::CGroundMoveType(CUnit* owner):
 
 	forceStaticObjectCheck = true;
 
+	flatFrontDir = (owner->frontdir * XZVector).Normalize();
```

Comment:
looks unrelated?

Reply:
It is completely related. FlatFrontDir has to be initialized. If it isn't and a new unit tries to move without trying to turn, then it will not animate correctly and it will not move at the correct speed.